### PR TITLE
Fix #125: agent settings UI padding when 'custom' agent selected

### LINKED ISSUE #125: UI strangeness in Agent Settings
### Intelligent Terminal version

0.1.1491.0

### Windows build number

_No response_

### Other Software

_No response_

### Steps to reproduce

Open settings > add new

### Expected Behavior

_No response_

### Actual Behavior

1. Padding, UI is off in Agent settings:

Image

2. Change ACP to the verbose "Agent Client Protocol":

Image

### PR DESCRIPTION
**Pull request by yeelam-gordon**
Fixes #125.

When a custom agent is selected for either the agent-pane agent or the command-palette agent, the SettingContainer card's content (ComboBox + command preview + Edit / Delete buttons) was flush against the top and bottom card edges.

The SettingContainer's NonExpanderGrid template has `Padding="16,0,8,0"` and `MinHeight=64`; the ContentPresenter is vertically centered. When the content grows past 64px the row grows to fit, but with no vertical padding the stacked rows visually hug the card border.

Added `Margin="0,12,0,12"` to the inner content StackPanel of both SettingContainers. The standard one-row layout (ComboBox alone) stays comfortably within `MinHeight=64`, so the fix is invisible there.

Issue #125 also asked to keep the 'ACP' acronym (not expand it to 'Agent Client Protocol'). That's already the case in en-US and every localized resw - no change needed.

### Validation
- Built CascadiaPackage locally (Debug, x64) - 0 errors.
- Visually verified both `AcpAgent` and `DelegateAgent` cards under `Settings > AI Agents` with a custom agent selected.
image

### CODE REVIEW

File: src/cascadia/TerminalApp/FreOverlay.cpp [hunk 1]
```diff
@@ -92,6 +95,56 @@ namespace winrt::TerminalApp::implementation
...
+                    const auto suffixStart = pos + acpToken.size();
+                    if (suffixStart < fullText.size())
+                    {
+                        Run suffix;
+                        suffix.Text(winrt::hstring{ fullText.substr(suffixStart) });
+                        inlines.Append(suffix);
+                    }
+                }
+            }
+        }
```
Comment (yeelam-gordon) [inline src/cascadia/TerminalApp/FreOverlay.cpp:187]: Looks like this whole code repeat twice.
There is no place for common code need for FRE and Terminal setting?

If no, no worry now.

Please also consider adding Unittest if possible here

Comment (yeelam-gordon) [reply src/cascadia/TerminalApp/FreOverlay.cpp:187]: Good catch — extracted the shared logic into `src/cascadia/inc/AcpLinkInjector.h` (header-only, same pattern as `RtlHelper.h` and `WtaProcess.h`). `FreOverlay::Initialize` and `AIAgents::_InjectAcpHelpTextLink` now both call `Microsoft::Terminal::AcpLink::InjectAcpLink(TextBlock)`. Net diff: **-1 line** (~50 duplicated lines collapsed into one call each, +101 in the new header). See 786632be9.

On the unit test: I've **skipped it intentionally**. The helper mutates a live WinRT `TextBlock` (sets `.Text`, mutates `.Inlines`, appends `Hyperlink`/`Run` elements) which requires a XAML runtime + UI-thread test host. `ut_app` is a TAEF console UT host with no XAML island, so a real test would need either:
1. a new XAML-hosted test fixture (significant infra for a 30-line helper), or
2. extracting a pure-string `SplitOnAcp(text) → (prefix, suffix)` and testing that — but that splits trivial logic from the actually-risky part (Inlines mutation, idempotency on re-Loaded), so the test would only cover `std::wstring::find`.

If the helper grows (more locked tokens, multiple links, conditional formatting) I'll revisit. For now the cost/benefit doesn't justify a XAML-host UT.

File: src/cascadia/TerminalSettingsEditor/AIAgents.cpp [hunk 2]
```diff
@@ -22,6 +61,12 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
         PageSubtitlePrefix().Text(RS_(L"AIAgents_PageSubtitlePrefix"));
         PageSubtitlePrivacyLink().Text(RS_(L"AIAgents_PageSubtitlePrivacyLink"));
+
+        // Hook Loaded on both SettingContainers so we can rewrite their
+        // template's HelpTextBlock once the localized HelpText TemplateBinding
+        // has populated it.
+        AcpAgent().Loaded({ this, &AIAgents::_InjectAcpHelpTextLink });
+        DelegateAgent().Loaded({ this, &AIAgents::_InjectAcpHelpTextLink });
```
Comment (yeelam-gordon) [inline src/cascadia/TerminalSettingsEditor/AIAgents.cpp]: Do we have this? I don't remember the subtitle for delegate agent have this???
And delegate agent never need to be run for ACP at all !

File: src/cascadia/inc/AcpLinkInjector.h [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+#include <string>
+#include <string_view>
+
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Windows.UI.Xaml.Controls.h>
+#include <winrt/Windows.UI.Xaml.Documents.h>
+
+namespace Microsoft::Terminal::AcpLink
+{
+    // Replace `target`'s text with the same content, but with the literal
```
Comment (yeelam-gordon) [inline src/cascadia/inc/AcpLinkInjector.h]: How come the code is link hacking?
Do you mean after the resource loaded from resw, if we want to display some word as hyperlink, we need to even change the UI tree?
No simpler method?

**Review by vanzue — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/Resources/zh-CN/Resources.resw [hunk 4]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/Resources/zh-CN/Resources.resw]: Verified via `git cat-file blob HEAD~2:<file>` — this locale file's first three bytes were `3C 3F 78` (`<?x`), not `EF BB BF`, so it never had a UTF-8 BOM in the repo (only the `en-US` resw files did). No BOM to restore here. The two files that genuinely lost a BOM (`TerminalApp` and `TerminalSettingsEditor` en-US) were restored in e7557c93f.

File: src/cascadia/TerminalApp/Resources/zh-TW/Resources.resw [hunk 5]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalApp/Resources/zh-TW/Resources.resw]: Verified via `git cat-file blob HEAD~2:<file>` — this locale file's first three bytes were `3C 3F 78` (`<?x`), not `EF BB BF`, so it never had a UTF-8 BOM in the repo (only the `en-US` resw files did). No BOM to restore here. The two files that genuinely lost a BOM (`TerminalApp` and `TerminalSettingsEditor` en-US) were restored in e7557c93f.

File: src/cascadia/TerminalApp/Resources/en-GB/Resources.resw [hunk 6]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalApp/Resources/en-GB/Resources.resw]: Verified via `git cat-file blob HEAD~2:<file>` — this locale file's first three bytes were `3C 3F 78` (`<?x`), not `EF BB BF`, so it never had a UTF-8 BOM in the repo (only the `en-US` resw files did). No BOM to restore here. The two files that genuinely lost a BOM (`TerminalApp` and `TerminalSettingsEditor` en-US) were restored in e7557c93f.

File: src/cascadia/TerminalSettingsEditor/Resources/zh-TW/Resources.resw [hunk 7]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/Resources/zh-TW/Resources.resw]: Verified via `git cat-file blob HEAD~2:<file>` — this locale file's first three bytes were `3C 3F 78` (`<?x`), not `EF BB BF`, so it never had a UTF-8 BOM in the repo (only the `en-US` resw files did). No BOM to restore here. The two files that genuinely lost a BOM (`TerminalApp` and `TerminalSettingsEditor` en-US) were restored in e7557c93f.

File: src/cascadia/TerminalSettingsEditor/AIAgents.cpp [hunk 8]
```diff
@@ -11,6 +11,9 @@
 #include "AIAgents.h"
 #include "AIAgents.g.cpp"
 
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::UI::Xaml::Documents;
 using namespace winrt::Windows::UI::Xaml::Navigation;
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/AIAgents.cpp:14]: This comment is stale relative to commit ec28ed394 — the inline `<Run x:Name=...>` elements added there are projected as `winrt::Windows::UI::Xaml::Documents::Run`, and the code-behind calls `AcpAgentDescriptionBefore().Text(...)` / `AgentDescriptionBefore().Text(...)` etc., which require both the `Documents.h` include and the `using namespace` declaration. The imports are no longer dead — they back the split-on-`ACP` Run population that replaced the standalone `HyperlinkButton`.

File: src/cascadia/TerminalApp/FreOverlay.cpp [hunk 9]
```diff
@@ -11,9 +11,12 @@
 #include "../inc/ShellIntegration.h"
 #include "../inc/RtlHelper.h"
 
+#include <winrt/Windows.UI.Xaml.Documents.h>
+
 using namespace winrt::Windows::Foundation;
 using namespace winrt::Windows::UI::Xaml;
 using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::UI::Xaml::Documents;
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalApp/FreOverlay.cpp:19]: This comment is stale relative to commit ec28ed394 — the inline `<Run x:Name=...>` elements added there are projected as `winrt::Windows::UI::Xaml::Documents::Run`, and the code-behind calls `AcpAgentDescriptionBefore().Text(...)` / `AgentDescriptionBefore().Text(...)` etc., which require both the `Documents.h` include and the `using namespace` declaration. The imports are no longer dead — they back the split-on-`ACP` Run population that replaced the standalone `HyperlinkButton`.

File: src/cascadia/TerminalSettingsEditor/AIAgents.cpp [hunk 10]
```diff
@@ -22,6 +25,17 @@
 
         PageSubtitlePrefix().Text(RS_(L"AIAgents_PageSubtitlePrefix"));
         PageSubtitlePrivacyLink().Text(RS_(L"AIAgents_PageSubtitlePrivacyLink"));
+
+        // The Agent card is hand-built (not a SettingContainer) so we can put
+        // the "Learn more about ACP" link in Column 0 directly below the
+        // description text. Re-use the existing AIAgents_AcpAgent resw keys
+        // (originally targeting SettingContainer's Header/HelpText DPs) via
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/AIAgents.cpp]: Resolved by trimming the comment in a6a0d0d47 — `DPs` no longer appears anywhere in the change.

File: src/cascadia/TerminalSettingsEditor/AIAgents.cpp [hunk 11]
```diff
@@ -22,6 +25,39 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
+        // ResourceLoader, so we set Run.Text from code).
+        {
+            const std::wstring_view desc{ RS_(L"AIAgents_AcpAgent/HelpText") };
+            constexpr std::wstring_view token{ L"ACP" };
+            const auto pos = desc.find(token);
+            if (pos != std::wstring_view::npos)
+            {
+                AcpAgentDescriptionBefore().Text(winrt::hstring{ desc.substr(0, pos) });
+                AcpAgentDescriptionAcpToken().Text(winrt::hstring{ token });
+                AcpAgentDescriptionAfter().Text(winrt::hstring{ desc.substr(pos + token.size()) });
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/AIAgents.cpp]: Good catch — fixed in 8ff67bd76 by storing the `hstring` first (`const auto descStr = RS_(...); const std::wstring_view desc{ descStr };`) so the view stays bound to a named local for the duration of the block.

File: src/cascadia/TerminalApp/FreOverlay.cpp [hunk 12]
```diff
@@ -92,6 +95,26 @@ namespace winrt::TerminalApp::implementation
...
+        // locale's resw via {Locked="ACP"}).
+        {
+            const std::wstring_view desc{ RS_(L"FreOverlay_AgentDescription/Text") };
+            constexpr std::wstring_view token{ L"ACP" };
+            const auto pos = desc.find(token);
+            if (pos != std::wstring_view::npos)
+            {
+                AgentDescriptionBefore().Text(winrt::hstring{ desc.substr(0, pos) });
+                AgentDescriptionAcpToken().Text(winrt::hstring{ token });
+                AgentDescriptionAfter().Text(winrt::hstring{ desc.substr(pos + token.size()) });
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalApp/FreOverlay.cpp]: Good catch — fixed in 8ff67bd76 (same pattern as the Settings page: `const auto descStr = RS_(...); const std::wstring_view desc{ descStr };`).

File: src/cascadia/inc/AcpLinkInjector.h [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+    // substring "ACP" turned into a hyperlink. Idempotent and defensive — a
+    // no-op when:
+    //   - `target` is null,
+    //   - text is empty (e.g. HelpText not yet bound),
+    //   - no "ACP" substring is found (locale anomaly — leave text alone),
+    //   - a Hyperlink is already present (re-Loaded / re-Initialize firings).
+    inline void InjectAcpLink(const winrt::Windows::UI::Xaml::Controls::TextBlock& target)
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/inc/AcpLinkInjector.h]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

File: src/cascadia/TerminalSettingsEditor/AIAgents.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/AIAgents.cpp]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/AIAgents.cpp]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

File: src/cascadia/inc/AcpLinkInjector.h (hunk 3 above)
Comment (yeelam-gordon) [reply src/cascadia/inc/AcpLinkInjector.h]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

File: src/cascadia/TerminalSettingsEditor/Resources/qps-ploca/Resources.resw [hunk 14]
```diff
@@ -2762,7 +2762,11 @@
...
+  <data name="AIAgents_AcpReferenceLink.Content" xml:space="preserve">
+    <value>[!! Ļéàŕñ mōŕé åbōùţ ACP — extra padding to test layout overflow !!]</value>
+    <comment>{Locked="ACP"} HyperlinkButton displayed below the AIAgents_AcpAgent ComboBox that links to the Agent Client Protocol reference page.</comment>
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/Resources/qps-ploca/Resources.resw]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

File: src/cascadia/TerminalApp/Resources/qps-ploca/Resources.resw [hunk 15]
```diff
@@ -980,6 +980,10 @@ In this context, "agent" refers to an AI agent (e.g. Copilot, Claude, Gemini), n
     <value>Choose the agent used in the agent pane. Any compatible ACP agent will be detected.</value>
     <comment>{Locked="ACP"}
 In this context, "agent" refers to an AI agent (e.g. Copilot, Claude, Gemini), not a human agent or representative.</comment>
+  </data>
+  <data name="FreOverlay_AcpReferenceLink.Content" xml:space="preserve">
+    <value>[!! Ļéàŕñ mōŕé åbōùţ ACP — extra padding to test layout overflow !!]</value>
+    <comment>{Locked="ACP"} HyperlinkButton displayed below FreOverlay_AgentDescription that links to the Agent Client Protocol reference page.</comment>
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalApp/Resources/qps-ploca/Resources.resw]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

File: src/cascadia/TerminalSettingsEditor/AIAgents.xaml [hunk 16]
```diff
@@ -111,6 +116,13 @@
                 </StackPanel>
             </local:SettingContainer>
 
+            <!--  Learn more about ACP — placed as a sibling outside the
+                  SettingContainer so it sits left-aligned under the card,
+                  mirroring the Extensions_DisclaimerHyperlink pattern.  -->
+            <HyperlinkButton x:Uid="AIAgents_AcpReferenceLink"
+                             NavigateUri="https://aka.ms/intelligent-terminal-acpref"
+                             Margin="-12,0,0,0" />
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/AIAgents.xaml]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

File: src/cascadia/TerminalApp/FreOverlay.xaml [hunk 17]
```diff
@@ -216,6 +216,11 @@
                                     <TextBlock x:Uid="FreOverlay_AgentDescription"
                                                FontSize="12" Foreground="#99FFFFFF"
                                                TextWrapping="Wrap" />
+                                    <HyperlinkButton x:Uid="FreOverlay_AcpReferenceLink"
+                                                     NavigateUri="https://aka.ms/intelligent-terminal-acpref"
+                                                     Margin="-12,0,0,0"
+                                                     Padding="12,0,12,0"
+                                                     Foreground="#FF99EBFF" />
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalApp/FreOverlay.xaml]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

File: src/cascadia/TerminalSettingsEditor/AIAgents.xaml [hunk 18]
```diff
@@ -67,10 +67,66 @@
...
+                          description text above (HyperlinkButton's default
+                          is ~14, which renders visibly larger). The negative
+                          left margin + horizontal Padding cancels out the
+                          button's built-in padding so the link text aligns
+                          flush-left under the description.  -->
+                    <HyperlinkButton x:Uid="AIAgents_AcpReferenceLink"
+                                     NavigateUri="https://aka.ms/intelligent-terminal-acpref"
+                                     FontSize="12"
+                                     Margin="-12,2,0,0"
+                                     Padding="12,2,12,2" />
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/AIAgents.xaml]: Outdated by the current head (8ff67bd76) — the helper / extra-resw-key / HyperlinkButton wiring this targets has been replaced by an inline `<Run>` + `<Hyperlink>` populated from code-behind, with no new resw keys. Resolving.

**Review by vanzue — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/AIAgents.cpp [hunk 19]
```diff
@@ -22,6 +22,30 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
         PageSubtitlePrefix().Text(RS_(L"AIAgents_PageSubtitlePrefix"));
         PageSubtitlePrivacyLink().Text(RS_(L"AIAgents_PageSubtitlePrivacyLink"));
+
+        const auto agentHeader = RS_(L"AIAgents_AcpAgent/Header");
+        AcpAgentHeaderText().Text(agentHeader);
+
+        // Split the description on "ACP" (locked token) so it can be rendered as an inline Hyperlink.
+        {
+            const auto descStr = RS_(L"AIAgents_AcpAgent/HelpText");
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalSettingsEditor/AIAgents.cpp:31]: Already fixed in 8ff67bd76 — the `hstring` is now stored in a named local (`descStr`) before the view is taken, so it stays alive for the whole block.

File: src/cascadia/TerminalApp/FreOverlay.cpp [hunk 20]
```diff
@@ -164,6 +167,25 @@ namespace winrt::TerminalApp::implementation
         SettingsSubtitlePrefix().Text(RS_(L"FreOverlay_SettingsSubtitlePrefix"));
         SettingsSubtitleLink().Text(RS_(L"FreOverlay_SettingsSubtitleLink"));
 
+        // Split the description on "ACP" (locked token) so it can be rendered as an inline Hyperlink.
+        {
+            const auto descStr = RS_(L"FreOverlay_AgentDescription/Text");
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply src/cascadia/TerminalApp/FreOverlay.cpp:172]: Already fixed in 8ff67bd76 — the `hstring` is now stored in a named local (`descStr`) before the view is taken, so it stays alive for the whole block.
